File: day3/typescript/part_1.ts
import * as fs from 'fs';

const memory = fs.readFileSync('input.txt', 'utf8');

const regex = /mul\((\d+),(\d+)\)/g;
let result = 0;
let match: RegExpExecArray | null;

while ((match = regex.exec(memory)) !== null) {
  const x = parseInt(match[1], 10);
  const y = parseInt(match[2], 10);
  result += x * y;
}

console.log(result);

File: day3/java/part_1.java
import java.io.*;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.regex.*;

public class part_1 {
    public static void main(String[] args) {
        try {
            String memory = new String(Files.readAllBytes(Paths.get("input.txt")));

            Pattern pattern = Pattern.compile("mul\\((\\d+),(\\d+)\\)");
            Matcher matcher = pattern.matcher(memory);

            int result = 0;

            while (matcher.find()) {
                int x = Integer.parseInt(matcher.group(1));
                int y = Integer.parseInt(matcher.group(2));
                result += x * y;
            }

            System.out.println(result);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
